From: =?utf-8?q?Fabian_Kl=C3=B6tzl?= <fabian@kloetzl.info>
Date: Mon, 21 Dec 2015 13:39:45 +0100
Subject: added auto tools build system

Upstream did not provide a makefile or any more advanced build system. Thus,
I added files for autoconf and automake to help with the installation.
---
 Makefile.am  | 30 ++++++++++++++++++++++++++++++
 configure.ac | 23 +++++++++++++++++++++++
 2 files changed, 53 insertions(+)
 create mode 100644 Makefile.am
 create mode 100644 configure.ac

diff --git a/Makefile.am b/Makefile.am
new file mode 100644
index 0000000..dd11268
--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,30 @@
+bin_PROGRAMS= indelible
+indelible_SOURCES= src/indelible.cpp
+indelible_CPPFLAGS= -Isrc
+
+dist_man_MANS= indelible.1
+
+# The following files are #included by indelible.cpp
+EXTRA_indelible_SOURCES= src/control.cpp src/models.cpp src/paml.cpp src/randoms.cpp src/MersenneTwister.h
+
+nobase_dist_html_DATA= help/index.html help/examples.shtml help/manual.shtml \
+	help/example_files/AMINOACID.shtml help/example_files/basicaminoacid.shtml \
+	help/example_files/basiccodon.shtml help/example_files/basicnucleotide.shtml \
+	help/example_files/basic.shtml help/example_files/branch2.shtml \
+	help/example_files/branch.shtml help/example_files/codon_RATES.txt \
+	help/example_files/CODON.shtml help/example_files/CODONsites.shtml \
+	help/example_files/indels.shtml help/example_files/multi-partitions.shtml \
+	help/example_files/mylengthmodel.txt help/example_files/nucleotide-branch.shtml \
+	help/example_files/nucleotide_RATES.txt help/example_files/NUCLEOTIDE.shtml \
+	help/example_files/random_branch_length_trees.JPG \
+	help/example_files/rates.shtml help/example_files/settings.shtml \
+	help/example_files/sitebranch.shtml help/example_files/trees.shtml \
+	help/example_files/trees.txt help/example_files/userAAmodel.txt \
+	help/manual_files/branches.shtml help/manual_files/codon_RATES.txt \
+	help/manual_files/evolve.shtml help/manual_files/model.shtml \
+	help/manual_files/mylengthmodel.txt help/manual_files/nucleotide_RATES.txt \
+	help/manual_files/partitions.shtml \
+	help/manual_files/random_branch_length_trees.JPG \
+	help/manual_files/settings.shtml help/manual_files/symQmatrix.gif \
+	help/manual_files/tree.shtml help/manual_files/trees.txt \
+	help/manual_files/type.shtml help/manual_files/userAAmodel.txt
diff --git a/configure.ac b/configure.ac
new file mode 100644
index 0000000..3d34e72
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,23 @@
+AC_INIT([indelible], [1.03])
+AM_INIT_AUTOMAKE([-Wall foreign])
+
+AC_PROG_CPP
+AC_PROG_CXX
+
+AC_LANG(C++)
+
+AC_CHECK_LIB([m],[cos])
+
+AC_TYPE_SIZE_T
+AC_TYPE_INT32_T
+AC_TYPE_UINT32_T
+AC_HEADER_STDBOOL
+
+AC_FUNC_MALLOC
+AC_CHECK_HEADERS([limits.h])
+AC_CHECK_FUNCS([floor pow sqrt])
+
+AC_CONFIG_FILES([
+ Makefile
+])
+AC_OUTPUT
